tests/node/test_wakunode_filter.nim

```diff
@@ -0,0 +1,109 @@
+{.used.}  
+
+import
+  std/[options, tables, sequtils],
+  stew/shims/net as stewNet,
+  testutils/unittests,
+  chronos,
+  chronicles,
+  os,
+  libp2p/peerstore,
+  libp2p/crypto/crypto
+
+import
+  ../../../waku/waku_core,
+  ../../../waku/node/peer_manager,
+  ../../../waku/node/waku_node,
+  ../../../waku/waku_filter_v2,
+  ../../../waku/waku_filter_v2/client,
+  ../../../waku/waku_filter_v2/subscriptions,
+  ../testlib/common,
+  ../testlib/wakucore,
+  ../testlib/wakunode,
+  ../testlib/testasync,
+  ../testlib/futures
+
+let FUTURE_TIMEOUT = 1.seconds
+
+suite "Full Node - Waku Filter - End to End":
+  var client {.threadvar.}: WakuNode
+  var clientPeerId {.threadvar.}: PeerId
+  var server {.threadvar.}: WakuNode
+  var serverRemotePeerInfo {.threadvar.}: RemotePeerInfo
+  var pubsubTopic {.threadvar.}: PubsubTopicy
+  var contentTopic {.threadvar.}: ContentTopic
+  var contentTopicSeq {.threadvar.}: seq[ContentTopic]
+  var pushHandlerFuture {.threadvar.}: Future[(string, WakuMessage)]
+  var messagePushHandler {.threadvar.}: FilterPushHandler
+  
+  asyncSetup:
+    pushHandlerFuture = newFuture[(string, WakuMessage)]()
```

Comment:
It's a super nit but for consistency I would use here `newPushHandlerFuture()` like in the unsubscribed test case